crates/core/src/db/datastore/locking_tx_datastore/mut_tx.rs

```diff
@@ -349,42 +359,42 @@ impl MutTxId {
     }
 
     /// Retrieves or creates the insert tx table for `table_id`.
-    #[allow(clippy::type_complexity)]
+    #[allow(clippy::type_complexity, clippy::unnecessary_lazy_evaluations)]
```

Comment:
I am somewhat skeptical of this allow lint, especially at the function level. I'd ask:
1. Can this instead be allowed on an expression level?
2. Failing that, can you add a comment which calls out which specific expression is being allowed here?
3. In either case, can you comment why the lint is wrong?